flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherResourceCleanupTest.java

```diff
@@ -651,6 +661,104 @@ public void testFailingJobManagerRunnerCleanup() throws Exception {
         awaitStatus(dispatcherGateway, jobId, JobStatus.RUNNING);
     }
 
+    @Test(timeout = 5000L)
```

Comment:
We don't want to use timeouts in JUnit. This would enable us to capture timeouts in the CI pipeline which would print the thread dump at the end giving us some hints where the process halted.